### ISSUE TITLE
KBI check for p_emuldata fails

### ISSUE THREAD

**Issue by Sp1l** [HardenedBSD/hardenedBSD#268]
On hardened/11-stable/master  there's a build failure when the offset is checked.

https://github.com/HardenedBSD/hardenedBSD/blob/hardened/11-stable/master-libressl/sys/kern/kern_thread.c#L103

For now I've commented the check so I can build but it appears the proc struct we use on HardenedBSD has elements of different size so the offset is incorrect.

**Comment by lattera** [comment]
Are you using the HARDENEDBSD kernel?

**Comment by Sp1l** [comment]
Nope, it is a specific KERNCONF. I doubt that that has any impact here.

These asserts have been added in dad5ccf4808854af8f6af32ab4f15dca00dae42b to make the build fail if the proc struct in sys/proc.h was changed. See e.g. 4be195ca42f9809dadcafc6e7eb30d4b32827cde for updates to it.

I'll mail you the KERNCONF, make.conf, src.conf separately.

**Comment by lattera** [comment]
Having a custom kernel might have an impact if it lacks the PAX kernel option. Those asserts can only be hit if PAX isn't set:

https://github.com/HardenedBSD/hardenedBSD/blob/hardened/11-stable/master-libressl/sys/kern/kern_thread.c#L68-L75
